### ISSUE TITLE
Updating for OIDC

### ISSUE THREAD

**Issue by HoChiPants** [ahaydon/Qlik-Cli-Windows#193]
Is there any update on being able to create a virtual proxy using OIDC?

**Comment by ahaydon** [comment]
Here is come code that will create a virtual proxy using OIDC.

```powershell
# get the name of the local machine
$hostname = ([System.Net.Dns]::GetHostEntry('localhost')).hostname

# get the Qlik Sense node object for this machine as a list
$engines = @(Get-QlikNode -Filter "hostname eq '$hostname'")

# define the virtual proxy config
$vp = @{
    prefix = 'azure';
    description = 'Azure OIDC';
    loadBalancingServerNodes = $engines;
    sessionCookieHeaderName = 'X-Qlik-Session-Azure'
    websocketCrossOriginWhiteList = $hostname
    sessionInactivityTimeout = 1800
    authenticationMethod = 5
    oidcConfigurationEndpointUri = 'https://login.microsoftonline.com/40272bbd-72b8-48ec-9d52-0522adbbdf21/v2.0/.well-known/openid-configuration'
    oidcClientId = 'changeme'
    oidcClientSecret = 'changeme'
    oidcRealm = 'AZURE'
    oidcAttributeSub = 'sub'
    oidcAttributeName = 'name'
    oidcAttributeGroups = 'groups'
    oidcAttributeEmail = 'upn'
    oidcAttributeClientId = 'appid'
    oidcAttributePicture = ''
    oidcScope = 'openid allatclaims profile email'
}

# convert the definition to JSON
$json = $vp | ConvertTo-Json -Compress -Depth 10

# Post the request to create the virtual proxy config
Invoke-QlikPost /qrs/virtualproxyconfig $json
```

**Comment by ahaydon** [comment]
This is now available in the v1.22.0-beta3 release and is available for testing, this can be downloaded from the Release section on GitHub or using the `-AllowPrerelease` switch when installing modules in PowerShell.
